lib/libstatcounters/riscv/statcounters_md.inc

```diff
@@ -36,7 +42,35 @@
 STATCOUNTER_ITEM(cycles, cycles, cycle)
 STATCOUNTER_ITEM(instructions, instructions, instret)
 RISCV_STATCOUNTER(time)
-/* TODO: add the remaining counters once we have implementations that support them. */
+// XXX THIS CURRENTLY HAS TO MATCH THE STATIC ALLOCATION IN RISCV-PK
+// TODO: see hpm_init() in riscv-pk/machine/minit.c
+STATCOUNTER_ITEM(redirect, redirect, hpmcounter3)
```

Comment:
@arichardson Should these be `RISCV_STATCOUNTER` (with an extended interface)? I don't really understand what the difference between the two is meant to be.

Also, do we want a few `STATCOUNTERS_GROUP_END()`'s in here?

Reply:
For what it's worth, in my initial implementation, these used a 2 params macro, so I think I'd be happy with making `RISCV_STATCOUNTER` be a 2 params macro.
I don't really have an opinion on the group end bit really... it's only for the human readable output, right? I am not sure what other applications we have for it at the moment, and I suspect that if we think of something useful, we would likely have a new set of events to track anyways?

Reply:
From my understanding of the intent behind (MIPS|RISCV)_STATCOUNTER I think adding a second argument and using it for all these makes sense. I'd suggest also adding some `STATCOUNTERS_GROUP_END()`s in sensible places (i.e. misc counters as one group, then the various bits of the memory hierarchy at whatever granularity you think makes sense)

Reply:
I agree changing RISCV_STATCOUNTER to 
```
#ifndef RISCV_STATCOUNTER
#define RISCV_STATCOUNTER(name, csr) STATCOUNTER_ITEM(name, name, csr)
#endif
```
would avoid some duplication here.

Reply:
what do you guys think of where I put the group ends?